Tools/llm-tool/README.md

```diff
@@ -60,3 +60,187 @@ Building in Release / optimizations will remove a lot of tail calls in the C++
 layer.  These lead to the stack overflows.
 
 See discussion here: https://github.com/ml-explore/mlx-swift-examples/issues/3
+
+## LoRA
+
+`llm-tool` provides an example LoRA driver based on:
+
+- https://github.com/ml-explore/mlx-examples/blob/main/lora/README.md
+
+This is an example of using MLX to fine-tune an LLM with low rank adaptation
+(LoRA) for a target task.[^lora] The example also supports quantized LoRA
+(QLoRA).[^qlora] The example works with Llama and Mistral style models
+available on Hugging Face.
+
+In this example we'll use the WikiSQL[^wikisql] dataset to train the LLM to
+generate SQL queries from natural language. However, the example is intended to
+be general should you wish to use a custom dataset.
+
+> Note: Some of the prompts have newlines in them which is difficult to achieve via running in Xcode.
+
+Running `llm-tool lora` will produce help:
+
+```
+SUBCOMMANDS:
+  train                   LoRA training
+  fuse                    Fuse lora adapter weights back in to original model
+  test                    LoRA testing
+  eval                    LoRA evaluation
+```
+
+### Training
+
+The first step will be training the LoRA adapter.  For this example we will 
+assume https://github.com/ml-explore/mlx-examples has been checked out into 
+your home directory -- please adjust as necessary.  This will use `lora/data`
```

Comment:
Maybe we should just package the `lora/data` in `mlx-swift-examples` to make this a bit easier.? It is only 300KB.

Reply:
Done